addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -67,43 +67,8 @@ def done(self):
                     picking.batch_id.id,
                     picking.batch_id.name))
 
-        picking_to_backorder = self.env['stock.picking']
-        picking_without_qty_done = self.env['stock.picking']
-        for picking in pickings:
-            if all([x.qty_done == 0.0 for x in picking.move_line_ids]):
-                # If no lots when needed, raise error
-                picking_type = picking.picking_type_id
-                if (picking_type.use_create_lots or picking_type.use_existing_lots):
-                    for ml in picking.move_line_ids:
-                        if ml.product_id.tracking != 'none':
-                            raise UserError(_('Some products require lots/serial numbers.'))
-                # Check if we need to set some qty done.
-                picking_without_qty_done |= picking
-            elif picking._check_backorder():
-                picking_to_backorder |= picking
-            else:
-                picking._action_done()
         self.write({'state': 'done'})
-        if picking_without_qty_done:
-            view = self.env.ref('stock.view_immediate_transfer')
-            wiz = self.env['stock.immediate.transfer'].create({
-                'pick_ids': [(4, p.id) for p in picking_without_qty_done],
-                'pick_to_backorder_ids': [(4, p.id) for p in picking_to_backorder],
-            })
-            return {
-                'name': _('Immediate Transfer?'),
-                'type': 'ir.actions.act_window',
-                'view_mode': 'form',
-                'res_model': 'stock.immediate.transfer',
-                'views': [(view.id, 'form')],
-                'view_id': view.id,
-                'target': 'new',
-                'res_id': wiz.id,
-                'context': self.env.context,
-            }
-        if picking_to_backorder:
-            return picking_to_backorder.action_generate_backorder_wizard()
-        return True
+        return self.picking_ids.button_validate()
```

Comment:
if the user cancel the backorer the batch picking is still marked as done.. fuck

Reply:
don't care atm